app/src/main/java/com/gathering/android/common/Composables.kt

```diff
@@ -231,5 +243,65 @@ fun ProgressBar(
     }
 }
 
+@Composable
+@Preview(showBackground = true)
+fun ShowProfilePicturePreviewWithString() {
+    ShowImage(imageUri = "")
+}
+
+@Composable
+fun ShowImage(imageUri: String?) {
+    Card(
+        modifier = Modifier
+            .padding(25.dp)
+            .clip(CircleShape)
+            .border(2.dp, Color.Gray, CircleShape),
+        colors = CardDefaults.cardColors(
+            containerColor = Color.White
+        )
+    ) {
+        val painter = if (imageUri.isNullOrEmpty()) {
+            painterResource(id = R.drawable.ic_person)
+        } else {
+            rememberAsyncImagePainter(model = "https://moyar.dev:8080/photo/${imageUri}")
+        }
+        Image(
+            contentScale = ContentScale.Crop,
+            painter = painter,
+            contentDescription = "Profile picture",
+            modifier = Modifier
+                .size(200.dp)
+                .background(Color.White, CircleShape)
+        )
+    }
+}
+
+@Composable
+fun ShowImage(imageUri: Bitmap?) {
+    Card(
+        modifier = Modifier
+            .padding(25.dp)
+            .clip(CircleShape)
+            .border(2.dp, Color.Gray, CircleShape),
+        colors = CardDefaults.cardColors(
+            containerColor = Color.White
+        )
+    ) {
+        val painter = if (imageUri == null) {
+            painterResource(id = R.drawable.ic_person)
+        } else {
+            rememberAsyncImagePainter(model = imageUri)
+        }
+        Image(
+            contentScale = ContentScale.Crop,
+            painter = painter,
+            contentDescription = "",
+            modifier = Modifier
+                .size(200.dp)
+                .background(Color.White, CircleShape)
+        )
+    }
```

Comment:
since these two methods are almost the same we need to do one of the followings:

- try to convert the birmap to a uri and call the first method from this method
OR
- make a third private method holding the same code and calling that from these two methods !! although we dont want the private method to accept Any. to avoid using Any we need the private method to have two variables instead one called bitmap and one called imageUri and based on which one was provided it does its job.

im pretty sure the first way works. the second way can get ugly.

Reply:
can i create ticket fro this later? i am sure anahid has also modified this function here so I need everything to be merged first.

Reply:
sure create a ticket and comment it here

Reply:
https://trello.com/c/odSk9za2/148-modify-showimage-function-in-composableskt